core/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -138,7 +142,7 @@ public TopDocs call() throws Exception {
                         return new TopDocs(totalHitCountCollector.getTotalHits(), Lucene.EMPTY_SCORE_DOCS, 0);
                     }
                 };
-            } else if (searchContext.searchType() == SearchType.SCAN) {
+            } else if (searchType == SearchType.SCAN) {
```

Comment:
right and that is great, but I was more thinking about the not using the scan context and its collector at all and just what happens from line 165 in this file.

I think this improvement is super and this PR should just get pushed and this can maybe done in a follow up PR?